### PR TITLE
feat(pie-monorepo): DSW-920 added pre-push eslint command

### PR DESCRIPTION
**Pull request by JoshuaNg2332**
## Describe your changes (can list changeset entries if preferable)

- Added `pre-push` eslint command

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: .husky/pre-push [hunk 1]
```diff
@@ -3,3 +3,6 @@
 
 # Checks branch name for Jira ticket id
 packages/tools/git-hooks-scripts/check-branch-name.js $(git symbolic-ref --short HEAD)
+
+# Checks the diff files for lint errors before allowing to push
+yarn eslint $(git diff --name-only main HEAD | grep -E '\.(ts|tsx|js|jsx)$' | xargs) --max-warnings 0
```
Comment (JoshuaNg2332) [inline .husky/pre-push:8]: Runs `eslint` on all files committed compared to main.

File: .husky/commit-msg [hunk 2]
```diff
@@ -17,6 +17,7 @@ fi
 # Initialise git-secrets configuration
 git-secrets --register-aws > /dev/null
 
-echo "Running git-secrets..."
+echo "Checking commit message for secrets..."
 # Scans the commit message.
 git-secrets --commit_msg_hook -- "$@"
+echo "Checking commit message for secrets...DONE"
```
Comment (fernandofranca) [inline .husky/commit-msg:23]: Might this be a leftover of some test?

Comment (JoshuaNg2332) [reply .husky/commit-msg:23]: I added this back in as I thought this was just a simple message that would help it be clearer as to what is happening.

**Comment by siggerzz** [comment]
More of a question for the team:

If we have pre-push lint checks, how do we feel about potentially removing the linting step from CI?

**Comment by sebastienrocks** [comment]
> More of a question for the team:
> 
> If we have pre-push lint checks, how do we feel about potentially removing the linting step from CI?
> 
> We obviously have a problem at the moment in that people can skip Husky checks via `--no-verify` / `HUSKY=0`, but if the motivation for this PR is to reduce the feedback loop and speed up our workflow, maybe it's worth looking at?

As long as we have some way to enforce linting, I'm happy with it.
We shouldn't skip husky checks anyways, so low risk for me

**Review by sebastienrocks — APPROVED** [review]

**Comment by xander-marjoram** [comment]
@siggerzz I'd be inclined to keep it in there, for now at least. We could probably combine it with the unit test step which should save some build time?

**Review by xander-marjoram — APPROVED** [review]
LGTM. Do you need a changeset file though?

**Comment by fernandofranca** [comment]
@siggerzz I'd rather have linting enabled on CI side, than only on dev side.

I can think of cases where I'd like to temporarily push a branch that couldn't not pass linting just for the sake of having my WIP backed up before performing radical changes

**Review by fernandofranca — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

**Pull request merged**